modules/core/src/main/java/org/apache/ignite/internal/processors/metric/MetricRegistry.java

```diff
@@ -115,12 +117,17 @@ public void reset() {
     }
 
     /**
-     * Register existing metrics in this group with the specified name.
+     * Register existing metrics in this group with the specified name. Note that the name of the metric must
+     * start with the name of the current registry it is registered into.
      *
      * @param metric Metric.
      */
     public void register(Metric metric) {
-        addMetric(metric.name(), metric);
+        T2<String, String> splitted = fromFullName(metric.name());
```

Comment:
Let's rewrite it to

```
        assert fromFullName(metric.name()).get1().equals(regName);

        addMetric(fromFullName(metric.name()).get2(), metric);
```

Because, asserts should be enabled only in testing environment.

Reply:
Done.